gapii/cc/vulkan_mid_execution.cpp

```diff
@@ -1152,6 +1153,26 @@ void VulkanSpy::EnumerateVulkanResources(CallObserver* observer) {
           opaqueSparseBound ? opaqueSparseBindCount : 0,
           opaqueSparseBound ? opaqueSparseBinds.data() : nullptr);
 
+      // If there are sparse residency bindings for this image, recreate them.
+      if (image.second->mSparseImageMemoryBindings.size() > 0) {
+        uint32_t count = image.second->mSparseImageMemoryBindings.size();
+        std::vector<VkSparseImageMemoryBind> imageSparseBinds;
+        // Put the bindings in chronological order.
+        std::vector<uint32_t> order_keys;
+        imageSparseBinds.reserve(count);
+        order_keys.reserve(count);
+        for (const auto& b : image.second->mSparseImageMemoryBindings) {
+          order_keys.push_back(b.first);
+        }
+        std::sort(order_keys.begin(), order_keys.end());
+        for (auto k : order_keys) {
+          imageSparseBinds.push_back(image.second->mSparseImageMemoryBindings[k]);
+        }
+        RecreateBindImageSparseMemoryBindings(observer, image.second->mDevice,
```

Comment:
I don't think this will do what you want with sparse residency. 
We try to read the entire image memory, then upload the entire image. Since we cannot copy the entire image, this will lead to undefined behavior.

Reply:
@AWoloszyn Just updated with commit: 45f9a91. For partially bound buffers/images, we only recover the bound areas.